### ISSUE TITLE
Turn kalite into an easily pluggable Django application

### ISSUE THREAD

**Issue by benjaoming** [learningequality/ka-lite#2952]
This is a goal rather than a singular bug or issue, and the end-goal can be tested... I would basically just want this to work....

```
$ pip install kalite
$ django-admin startproject mykalite
$ cd mykalite
$ nano settings.py  # Insert 'kalite' in INSTALLED_APPS, other features are called 'kalite.plugins.xyz'
$ nano urls.py  # Insert kalite's root url pattern
$ python manage.py syncdb  # Asks user to create a superuser
$ python manage.py migrate
$ python manage.py runserver  # And now kalite should run.
```

You're probably asking, why we need this since it doesn't introduce any brand new features or satisfy an obvious use case.. The answers are IMO:
1. Because kalite can improve its architecture by being closer to state-of-the-art Django and how applications (especially the bigger ones) are designed.
2. Because being like the rest of the ecosystem will most likely make it easier to include that system (so be compatible with other django stuff)
3. Because there is a potential open source community (unleashed?) with django expertice will find it intuitive to contribute to kalite if it behaved as a normal, pluggable application.

After having a more state-of-the-art Django application, we can now ask again: How do we deploy it on different platforms automatically? The answer to that is given a lot of places, because we've managed to refactor the problem (deployment of our software project) into a universal problem by making the application itself be universal. For instance, we can now start using [django project templates](https://github.com/django-wiki/django-wiki-project-template). Example:

```
$ django-admin.py startproject --template=https://github.com/learningequality/ka-lite-project-template/archive/master.zip <project_name>
```

Are there any real use cases, otherwise? Well, I'd like to deploy a handful of Django applications, and I would like to configure my own project environment for that, so I can run a CMS for a school, a Wiki, and kalite... all in the same server with the same UWSGI instance.

**Comment by aronasorman** [comment]
@benjaoming can you give a draft of the new folder structure and architecture you're proposing?

**Comment by benjaoming** [comment]
I would propose not refactoring applications, but only factoring out project stuff and settings.

    ka-lite/
        kalite_project/
            settings/
                applications/
                    __init__.py
                    some_application.py
                __init__.py
                base.py
                dev.py
                cherrypy.py
            manage.py
        kalite/
            some_application/
                __init__.py
                settings.py
            __init__.py

The purpose of `kalite_project` would be to hold a basic project structure that is both useful for development and deployment. If someone would need to create a different project, they could do so by importing everything from kalite_project.settings and overriding it with their own stuff.

**Comment by aronasorman** [comment]
We now can do a similar installation process, I believe:

```
pip install ka-lite-static
kalite manage setup
kalite start
```

Can we close this @benjaoming?

**Comment by aronasorman** [comment]
Although we still don't have that template functionality you put in the description. Maybe open another issue for that?

**Comment by benjaoming** [comment]
No @aronasorman, this feature is far from implemented, wouldn't dream of closing this issue yet :)

**Comment by benjaoming** [comment]
We went some of the way on this, but keeping this issue open won't help us much.

One of the prerequisites were #4054 and #4191 which were never fully solved.

We did, however, manage to get some of the project-related stuff isolated, and it still seems reasonable to try to centralize+document all project-wide settings.

Finally, as we don't encourage any new+advanced deployments of KA Lite, it doesn't seem very feasible to solve this issue.
